docs/Dialects/DC/RationaleDC.md

```diff
@@ -0,0 +1,43 @@
+
+The intention of DC is to model all operations required to represent such a control flow language. DC aims to be strictly a control
+flow dialect - as opposed to the Handshake dialect, which assigns control semantics to _all_ SSA values. As such, data values are
+only present in DC where they are required to model control flow.
+
+By having such a control language, the dialect aims to facilitate the construction of dataflow programs where control and data
+is explicitly separate. This enables optimization of the control and data side of the program independently, as well as mapping
+of the data side into functional units, pipelines, etc.. Furthermore, separating data and control will make it easier to reason
+about possible critical paths of either circuit, which may inform buffer placement.
+
+The DC dialect has been heavily influenced by the Handshake dialect, and can either be seen as a successor to it, or a lower level
+abstraction. As of writing, DC is _fully deterministic_. This means that non-deterministic operators such as the ones found in Handshake - `handshake.merge, handshake.control_merge` - do **not** have a lowering to `dc`. `handshake` programs must therefore be converted or by construction not contain any of these non-deterministic operators. Apart from that, **all** handshake operations can be lowered to a combination of DC
+and e.g. `arith` operations (to represent the data side semantics of any given operation).
+
+
+## Operations
+In `dc`, there are two main classes of operations:
+1. **Control-only operations**: Control-only operations come in pairs and represent start- and end-points in the control network.
+  * `fork/join`: A fork splits the incoming token into multiple outgoing tokens - a join is the reverse, wherein it synchronizes all incoming tokens into a single outgoing token.
+2. Domain-crossing operations: that is, operations which carry both control- and data semantics.
+   * `dc.buffer` is an operation which may buffer an arbitrary amount of data values, and is controlled by a `dc.token` transaction.
+   **Hardware note**: buffers have no dialect-side notion of cycles/stages/implementation. It is up to the generating pass to interpret buffer semantics - some may want to add attributes to a single buffer, some may want to stagger `dc.buffer`s sequentially.
+   * `branch/select`: Notably, a branch and select op may have **at most** 2 out/inputs. In `dc`, determism is assumed, and by extension, a `select` will always *eventually* be driven by the input of a corresponding `branch` operation. A `branch` operation is expected to be driven by a data-side `i1` value.
+     * A `branch` will transact once its select input is valid, the input is valid, and the selected output is ready. **Only** the selected output will be transacted.
+     * Conversely, a `select` will transact once its select input is valid, the selected input is valid, and the output is ready.
```

Comment:
The level of granularity in the `branch/select` descriptions seems a little low-level compared to `fork/join`, `pack/unpack`. Maybe move this prose to the operation's descriptions in ODS? I don't think we need the detailed semantics of the operations in this document, it seems like you are just trying to enumerate some of the different kinds of operations.